theano/scalar/basic.py

```diff
@@ -1151,7 +1151,13 @@ def c_code(self, node, name, inputs, outputs, sub):
         (z,) = outputs
         if node.inputs[0].type in complex_types:
             raise NotImplementedError()
-        return "%(z)s = isnan(%(x)s);" % locals()
+        # Windows tries to be different and sometimes return -1, but we want
+        # to be consistent with numpy (which returns True), hence the "abs".
+        return "%(z)s = abs(isnan(%(x)s));" % locals()
```

Comment:
this would cause slow down on mac and linux. Can you add the abs only on windows?